cmd/cluster/none/create.go

```diff
@@ -67,7 +67,10 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	// TODO: We are inconsistent atm as DNS basedomain input is required and platform agnostic at the HostedCluster API level.
 	// However In the CLI input is only exposed in AWS and not required atm. This should either be core input and required for
 	// every platform or the field in the API should be optional if there's platform which don't need it.
-	exampleOptions.BaseDomain = "example.com"
+	exampleOptions.BaseDomain = opts.BaseDomain
+	if exampleOptions.BaseDomain == "" {
+		exampleOptions.BaseDomain = "example.com"
```

Comment:
should this be defaulted [here](https://github.com/openshift/hypershift/pull/821/files#diff-6ff08b3e8d7fd9373119b5ac911e6f0e4e7e054905d982423ce9414577a53352R37) instead?

Reply:
I think it is in general fine this way since as we see in the case of AWS, there can be fallback paths to find the value. In some cases, with the cobra library, I used a default value and then checked if a user supplied the default value or if the default is used (can be seen on the flags), but it has similar complexity.